File: solutions/go/0206-reverse-linked-list.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
	var reversed *ListNode

	for head != nil {
		head.Next, head, reversed = reversed, head.Next, head
	}

	return reversed
}

File: solutions/php/0206-reverse-linked-list.php
<?php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution
{
    /**
     * @param ListNode $head
     * @return ListNode
     */
    function reverseList(?ListNode $head): ?ListNode
    {
        $reversed = null;

        while ($head) {
            [$head->next, $head, $reversed] = [$reversed, $head->next, $head];
        }

        return $reversed;
    }
}
